compiler/rustc_passes/src/stability.rs

```diff
@@ -828,7 +828,15 @@ impl Visitor<'tcx> for Checker<'tcx> {
 
     fn visit_path(&mut self, path: &'tcx hir::Path<'tcx>, id: hir::HirId) {
         if let Some(def_id) = path.res.opt_def_id() {
-            self.tcx.check_stability(def_id, Some(id), path.span, None)
+            let method_span = if path.segments.len() >= 2 {
```

Comment:
Maybe other logic is needed here? Only for method? But I'm not sure how to do that.